source/vsm/vsm/db/sqlalchemy/api.py

```diff
@@ -4305,21 +4287,14 @@ def get_poolusage(context, poolusage_id):
     return result
 
 def latency_all_types(context, search_opts, session=None):#for latency
+    # Get diamond collection interval
```

Comment:
I see the same logic appears a few time, why not to extract the logic into a function?